django_python3_ldap/tests.py

```diff
@@ -247,3 +261,146 @@ def testImportFunc(self):
 
         with self.settings(LDAP_AUTH_SYNC_USER_RELATIONS='django.contrib.auth.get_user_model'):
             self.assertTrue(callable(import_func(settings.LDAP_AUTH_SYNC_USER_RELATIONS)))
+
+    # User cleaning.
+
+    def testCleanUsersDeactivate(self):
+        from django.contrib.auth import get_user_model
+        User = get_user_model()
+        _username = "nonldap{user}".format(user=settings.LDAP_AUTH_TEST_USER_USERNAME)
+        user = User.objects.create_user(
+            _username,
+            "nonldap{mail}".format(mail=settings.LDAP_AUTH_TEST_USER_EMAIL),
+            settings.LDAP_AUTH_TEST_USER_PASSWORD)
+        user.save()
+        user_count_1 = User.objects.count()
+        self.assertEqual(User.objects.get(username=_username).is_active, True)
+        call_command("ldap_clean_users", verbosity=0)
+        user_count_2 = User.objects.count()
+        self.assertEqual(user_count_1, user_count_2)
+        self.assertEqual(User.objects.get(username=_username).is_active, False)
+
+        ### Test with lookup
+        # Reactivate user
+        user = User.objects.get(username=_username)
+        user.is_active = True
+        user.save()
+        # Create second user
+        _usernameLookup = "nonldaplookup{user}".format(user=settings.LDAP_AUTH_TEST_USER_USERNAME)
+        user = User.objects.create_user(
+            _usernameLookup,
+            "nonldaplookup{mail}".format(mail=settings.LDAP_AUTH_TEST_USER_EMAIL),
+            settings.LDAP_AUTH_TEST_USER_PASSWORD)
+        user.save()
+        user_count_1 = User.objects.count()
+        self.assertEqual(User.objects.get(username=_usernameLookup).is_active, True)
+        # Clean second user
+        call_command("ldap_clean_users", _usernameLookup, verbosity=0)
+        user_count_2 = User.objects.count()
+        self.assertEqual(user_count_1, user_count_2)
+        self.assertEqual(User.objects.get(username=_usernameLookup).is_active, False)
+        self.assertEqual(User.objects.get(username=_username).is_active, True)
+        # Reactivate second user
+        user = User.objects.get(username=_usernameLookup)
+        user.is_active = True
+        user.save()
+        # Clean first user
+        call_command("ldap_clean_users", _username, verbosity=0)
+        self.assertEqual(User.objects.get(username=_username).is_active, False)
+        self.assertEqual(User.objects.get(username=_usernameLookup).is_active, True)
+        # Lookup a non existing user (raise a CommandError)
+        try:
+            call_command("ldap_clean_users", 'doesnonexist', verbosity=0)
+        except Exception as e:
```

Comment:
You should use `assertRaises(CommandError)` here, rather than catching and printing the error.

Reply:
Thank you for this tip, it is just what I am looking for